### PR TITLE
Makisheva/rich input sanitization

### PR DESCRIPTION
**Pull request by AlbinaMakisheva**

### CODE REVIEW

File: ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx [hunk 1]
```diff
@@ -320,6 +332,11 @@ const FormFieldsContainer = () => {
               textFieldLabel={i18next.t("Description")}
               fieldPath="metadata.abstract"
               rich={true}
+              editorConfig={{
+                toolbar:
+                  "bold italic | bullist numlist | outdent indent | undo redo",
+                valid_elements: "strong,b,div,br,p,i,li",
```
Comment (mirekys) [inline ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx:338]: Inconsistent with https://github.com/oarepo/oarepo-ui/pull/184

Comment (mirekys) [reply ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx:338]: List of valid tags should come from a single source of truth, do not copy it in every possible input

File: ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx [hunk 2]
```diff
@@ -343,11 +360,22 @@ const FormFieldsContainer = () => {
...
+            <MultilingualTextInput
+              labelIcon="pencil"
+              label={i18next.t("Notes")}
+              textFieldLabel={i18next.t("Note")}
               fieldPath="metadata.notes"
+              rich={true}
+              editorConfig={{
+                toolbar:
+                  "bold italic | bullist numlist | outdent indent | undo redo",
+                valid_elements: "strong,b,div,br,p,i,li",
```
Comment (mirekys) [inline ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx:372]: List of valid tags should come from a single source of truth, do not copy it in every possible input

**Review by mirekys — CHANGES_REQUESTED** [review]

File: ui/branding/semantic-ui/less/theme.config
Comment (mirekys) [inline ui/branding/semantic-ui/less/theme.config:1]: Revert the changes to the file, we'd like to keep the spacing

File: ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx [hunk 3]
```diff
@@ -37,6 +39,11 @@ const FormFieldsContainer = () => {
     []
   );
 
+  const { values, setFieldValue, setFieldTouched } = useFormikContext();
+
+  const validTags = "strong,b,div,br,p,i,li";
```
Comment (mirekys) [inline ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx:44]: Is there any reason to not to use the default accepted tags set, consistent with what backend also allows?

**Review by mirekys — CHANGES_REQUESTED** [review]

File: ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx [hunk 4]
```diff
@@ -37,6 +39,12 @@ const FormFieldsContainer = () => {
     []
   );
 
+  const { values, setFieldValue, setFieldTouched } = useFormikContext();
+
+  const validTags =
+    "a,abbr,acronym,b,blockquote,br,code,div,table,tbody,td,th,tr,em,h1,h2,h3,h4,h5,i,li,ol,p,pre,span,strike,strong,sub,sup,u,ul";
```
Comment (mirekys) [inline ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx:45]: Is there any difference between this and [this](https://github.com/oarepo/oarepo-ui/pull/188/files#diff-7faf2dfbd5b15c265ec6341bbbee21521b1d5faaea7e871e5036f06fdc6f5f88R30) ?

**Review by mirekys — CHANGES_REQUESTED** [review]

File: ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx (hunk 4 above)
Comment (mirekys) [reply ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx]: Perfect, not it's exactly the same

Comment (mirekys) [reply ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx]: Now what is the added value of exactly duplicating this set of tags, if it's the default value for sanitize function?

Comment (AlbinaMakisheva) [reply ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx]: Tiny-mce expects it in the different format than we have it for sanitisation function

File: ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx [hunk 5]
```diff
@@ -4,6 +4,8 @@ import {
   MultilingualTextInput,
   FormikStateLogger,
   EDTFSingleDatePicker,
+  sanitizeInput,
+  validTags
```
Comment (mirekys) [inline ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx:8]: Screenshot 2024-06-17 at 10 13 44

Is it really exported in some PR?

Comment (AlbinaMakisheva) [reply ui/documents/semantic-ui/templates/documents/form/FormFieldsContainer.jsx:8]: https://github.com/oarepo/oarepo-ui/pull/194

**Review by mirekys — APPROVED** [review]
